src/trainer.py

```diff
@@ -1093,19 +1062,10 @@ def _save_checkpoint(self, training_state, phase="pretrain", new_best=False, tas
             task_states[task_name] = {}
             task_states[task_name]["total_batches_trained"] = task_info["total_batches_trained"]
             task_states[task_name]["stopped"] = task_info["stopped"]
-            if self._g_optimizer is None:
-                task_states[task_name]["optimizer"] = task_info["optimizer"].state_dict()
-                sched_params = {}
-                task_states[task_name]["scheduler"] = sched_params
         task_states["global"] = {}
-        task_states["global"]["optimizer"] = (
-            self._g_optimizer.state_dict() if self._g_optimizer is not None else None
-        )
-        if self._g_scheduler is not None:
-            sched_params = {}
-            task_states["global"]["scheduler"] = sched_params
-        else:
-            task_states["global"]["scheduler"] = None
+        task_states["global"]["optimizer"] = self._optimizer.state_dict()
+        # NOTE(Alex): AllenNLP wrapper doesn't expose scheduler state dict methods
```

Comment:
Hrm?

Reply:
Just to explain why underneath we have to do `self._scheduler.lr_scheduler`